doc/guides/porting/arch.rst

```diff
@@ -441,13 +441,10 @@ However, a real implementation is strongly recommended.
 Fault Management
 ****************
 
-Each architecture provides two fatal error handlers:
-
-* :code:`_NanoFatalErrorHandler`, called by software for unrecoverable errors.
-* :code:`_SysFatalErrorHandler`, which makes the decision on how to handle
-  the thread where the error is generated, most likely by terminating it.
-
-See the current architecture implementations for examples.
+In the event of an unhandled CPU exception, the architecture code must call
+into :c:func:`z_fatal_error`. If the platform supports it, it may optionally
```

Comment:
This could use an explanation of what z_fatal_error does.   The ``z_fatal_error`` API doxygen comments don't mention that it could call ``z_arch_system_halt``, but do talk about calling ``k_sys_fatal_error_handler()``. 
```
 * This function dumps out architecture-agnostic information about the error
 * and then makes a policy decision on what to do by invoking
 * k_sys_fatal_error_handler().
 *
```
How is the optional behavior configured for spinning or powering off?

Reply:
@dbkinder I'm not sure I understand your review.
The implementation of k_sys_fatal_error_handler(), which in most cases will be overridden by the default implementation provided here, completely determines what happens. It could do anything it likes.

Reply:
> How is the optional behavior configured for spinning or powering off?

It's literally the k_sys_fatal_error_handler() function body.
Note that is a weak function, which can be overridden by applications to implement their own policy.

Reply:
Ah.  So how about something like this:
```
In the event of an unhandled CPU exception, the architecture
code must call into :c:func:`z_fatal_error`.  This function dumps
out architecture-agnostic information and makes a policy
decision on what to do next by invoking ``k_sys_fatal_error()``.
This function can be overridden to implement application-specific
policies that could include locking interrupts and spinning forever
(the default implementation) or even powering off the
system (if supported).
```

Reply:
Thanks, will do

Reply:
done